### PR TITLE
Sdp overhaul

### PR DESCRIPTION
**Pull request by bbaldino**

### CODE REVIEW

**Comment by damencho** [comment]
Jenkins, add to whitelist, please.

**Comment by paweldomas** [comment]
The test have failed:
22:39:07 modules/xmpp/sdp-consistency.js: line 63, col 2, Missing semicolon.
22:39:07 modules/xmpp/sdp-consistency.js: line 67, col 2, Missing semicolon.
22:39:07 
22:39:07 2 errors

**Comment by bbaldino** [comment]
this depends on https://github.com/jitsi/lib-jitsi-meet/pull/338

File: conference.js [hunk 1]
```diff
@@ -960,31 +957,27 @@ export default {
...
-                APP.UI.addLocalStream(stream);
-            } else {
-                this.audioMuted = false;
-            }
-
-            APP.UI.setMicrophoneButtonEnabled(true);
-            APP.UI.setAudioMuted(this.getMyUserId(), this.audioMuted);
-        });
+    useAudioStream (newStream) {
+        room.replaceStream(localAudio, newStream)
```
Comment (hristoterezov) [inline conference.js]: I think now we are not returning anything anymore. Is this by mistake?

We should either return Promise either fix the documentation if we don't need the Promise anymore.

File: conference.js [hunk 2]
```diff
@@ -915,43 +915,40 @@ export default {
...
-                this.isSharingScreen = false;
-            }
+  /**
+   * Start using provided video stream.
+   * Stops previous video stream.
+   * @param {JitsiLocalTrack} [stream] new stream to use or null
+   * @returns {Promise}
+   */
+    useVideoStream (newStream) {
+        room.replaceStream(localVideo, newStream)
```
Comment (hristoterezov) [inline conference.js]: I think now we are not returning anything anymore. Is this by mistake?

We should either return Promise either fix the documentation if we don't need the Promise anymore.

Comment (bbaldino) [inline conference.js]: fixed (returned the promise)

File: conference.js (hunk 1 above)
Comment (bbaldino) [inline conference.js]: fixed (returned the promise)

**Pull request merged**